### PR TITLE
Limiting investments to 1 bil

### PR DESCRIPTION
**Pull request by thecsw**
At the request of r/MemeEconomy mods, this update limit the amount of memecoins users can invest to just 1 Billion.

### CODE REVIEW

**Comment by luisbg** [comment]
With firms losing 10% every week. This will make the big ones lose net worth faster than they can get it back. Is that the intention?

**Comment by keanucz** [comment]
That is the intention as far as I see. They will lower the skill ceiling for firms, bring new ones, and diversify the pool of memes rather than a low-effort shitshow.

**Comment by osmarks** [comment]
What if people just invest twice?

File: src/comment_worker.py [hunk 1]
```diff
@@ -283,6 +283,9 @@ def invest(self, sess, comment, investor, amount, suffix):
         if amount < minim or amount < 100:
             return comment.reply_wrap(message.modify_min_invest(minim))
 
+        if amount > 1e9:
+            return comment.reply_wrap("Investing limit is 1 billion")
```
Comment (luisbg) [inline src/comment_worker.py]: Wouldn't it be better to accept the investment but cap it?

`if amount > 1e9:
    amount = 1e9
`

**Comment by Winb23** [comment]
W H A T
W H Y

File: src/comment_worker.py (hunk 1 above)
Comment (osmarks) [reply src/comment_worker.py]: It probably should provide some notice, though.

**Comment by thomasvt1** [comment]
This would still allow multiple comments investing 1b. Probally needs some check to see if the user already invested

**Comment by Winb23** [comment]
why are we doing this though

**Comment by keanucz** [comment]
@Winb23 To encourage creativity and throw off the firm monopoly. The big firms will go bankrupt because they can't recoup their net worth.

**Review by hazzakak — COMMENTED** [review]
Should have if it's 100%, set it to 1b

**Review by keanucz — CHANGES_REQUESTED** [review]
You should probably make sure they've already invested on the post or otherwise more load on reddit, more comment spam.

**Comment by osmarks** [comment]
@Keanu73 There are better ways to do that though without as many negative effects on stuff.

**Comment by Winb23** [comment]
@Keanu73 you guys might want to run this by the actual investors first

**Comment by luisbg** [comment]
My only worry about this is the unfair advantage that big investors with huge net worth now have. Nobody can catch up to them.

**Comment by hazzakak** [comment]
It should probably be introduced in a new season.

**Comment by Winb23** [comment]
@luisbg my concern is how much fucking backlash this is going to cause

**Comment by luisbg** [comment]
It can be announced as a sticky post in the sub. So it doesn't catch people by surprise.

**Comment by ThoughtVendor** [comment]
So is this how you kill a sub? Lol

**Comment by Dracologist** [comment]
Make sure you limit how many times you can invest in a single post or all this will do is cause more comment spam and bot lag

**Comment by Dracologist** [comment]
Perhaps it would be better to apply a percentage cap to investments above a certain threshold. For example, what if you couldn't invest more than 50% of your balance or 1 billion MemeCoins (whichever is higher) in a single meme?

File: src/comment_worker.py (hunk 1 above)
Comment (soleary1222) [reply src/comment_worker.py]: maybe it could do both?  accept it and add the comment?

**Review by soleary1222 — COMMENTED** [review]
I like this approach much better.  But wouldn't people just invest twice in the same meme causing more load on the bot?

**Comment by thecsw** [comment]
> I like this approach much better. But wouldn't people just invest twice in the same meme causing more load on the bot?

No, you see, imagine you have 1000 Mc. Investing 50% would invest 500, leaving you with 500. If you try to invest 50% again, it would invest 250, leaving you with 250, and etc. Basically, it kinda limits people.

Don't worry about the load on the bot. He can handle himself

**Comment by ThoughtVendor** [comment]
So the idea is to encourage constant investments until you have under 1k MC? If so, keep in mind the volume of posts. I feel like this could just cause an influx of massive shitposts to meet the needs to max invest. Basically people would invest an insane amount of times this way and encourage them to automatically up vote everything that gets posted. So this may swing the pendulum the wrong direction. 

Might make more sense to design it so that you can only have 4 active investments at a time. First investment at 25%. Second at 50%. Third at 75%. Fourth at 100%. 

This would likely rotate active investors having to constantly invest 1/4th of their coins about every hour. Which is a bit unrealistic...especially because of the return dropoff after 5 upvotes...

**Comment by hazzakak** [comment]
Yeah unless you can only invest on a post once then the bot won't have 700 people commenting on a post in a minute.

**Comment by Dracologist** [comment]
Can we make it so that the 50% cap only applies to investments above a certain amount? This cap could be very punishing for the little guy

EDIT: I wrote up some code that should do that. Check the pull requests

**Comment by thecsw** [comment]
@Dracologist Please see the updates on this PR.

**Comment by thecsw** [comment]
I am closing this PR. See #401 for the current updates on the new feature.

**Pull request closed without merging**